core/src/query.rs

```diff
@@ -1,77 +1,551 @@
+        self.current = Some(self.resultset_cursor.deserialize_current()?);
+        res
     }
 
     // Get the BSON value for the cell at the given colIndex on the current row.
     // Fails if the first row as not been retrieved (next must be called at least once before getValue).
-    fn get_value(&self, _col_index: u16) -> Result<Option<Bson>> {
-        unimplemented!()
+    fn get_value(&self, col_index: u16) -> Result<Option<Bson>> {
+        let current = self.current.as_ref().ok_or(Error::InvalidCursorState)?;
+        let md = self.get_col_metadata(col_index)?;
+        let datasource = current
+            .get_document(&md.table_name)
+            .map_err(Error::ValueAccess)?;
+        let column = datasource.get(&md.col_name);
+        Ok(column.cloned())
     }
 }
 
 // Metadata information for a column of the result set.
 // The information is to be used when reporting columns information from
-// SQLColAtrribute or SQLDescibeCol and when converting the data to the targeted C type.
-#[derive(Debug)]
+// SQLColAttribute or SQLDescribeCol and when converting the data to the targeted C type.
+#[derive(Clone, Debug)]
 pub struct MongoColMetadata {
     pub base_col_name: String,
     pub base_table_name: String,
     pub catalog_name: String,
-    pub col_count: u16,
-    pub display_size: u64,
+    pub display_size: Option<u16>,
     pub fixed_prec_scale: bool,
     pub label: String,
-    pub length: u128,
+    pub length: Option<u16>,
     pub col_name: String,
-    pub is_nullable: bool,
-    pub octet_length: u128,
-    pub precision: u16,
-    pub scale: u16,
+    pub is_nullable: ColumnNullability,
+    pub octet_length: Option<u16>,
+    pub precision: Option<u16>,
+    pub scale: Option<u16>,
     pub is_searchable: bool,
     pub table_name: String,
     // BSON type name
     pub type_name: String,
     pub is_unsigned: bool,
     pub is_updatable: bool,
 }
+
+#[derive(Clone, Debug)]
+pub enum ColumnNullability {
+    Nullable,
+    NoNulls,
+    Unknown,
+}
+
+// Struct representing the response for a sqlGetResultSchema command.
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone, Default)]
+struct SqlGetResultSchemaResponse {
+    pub ok: i32,
+    pub schema: VersionedJsonSchema,
+}
+
+// Auxiliary struct representing part of the response for a sqlGetResultSchema
+// command.
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone, Default)]
+struct VersionedJsonSchema {
+    pub version: i32,
+    pub json_schema: json_schema::Schema,
+}
+
+// A simplified JSON Schema, relative to the json_schema::Schema struct.
+// An instance of SimplifiedJsonSchema is semantically equivalent to its
+// corresponding json_schema::Schema, but with two main simplifications.
+// 1. The bson_type has to be a single type. If the json_schema::Schema
+// contains multiple bson_types, they are pushed down into the any_of list.
+// 2. The any_of is flattened.
+#[derive(Clone, Default)]
+struct SimplifiedJsonSchema {
```

Comment:
This is _roughly_ equivalent to `MongoJsonSchema` from the JDBC driver. See [that class](https://github.com/mongodb/mongo-jdbc-driver/blob/master/src/main/java/com/mongodb/jdbc/MongoJsonSchema.java) to jog your memory.